bgpd/bgp_zebra.c

```diff
@@ -3155,26 +3155,23 @@ static int bgp_zebra_process_srv6_locator_chunk(ZAPI_CALLBACK_ARGS)
 	struct stream *s = NULL;
 	struct bgp *bgp = bgp_get_default();
 	struct listnode *node;
-	struct prefix_ipv6 *c;
-	struct srv6_locator_chunk s6c = {};
-	struct prefix_ipv6 *chunk = NULL;
+	struct srv6_locator_chunk *c;
+	struct srv6_locator_chunk *chunk = srv6_locator_chunk_alloc();
 
 	s = zclient->ibuf;
-	zapi_srv6_locator_chunk_decode(s, &s6c);
+	zapi_srv6_locator_chunk_decode(s, chunk);
 
-	if (strcmp(bgp->srv6_locator_name, s6c.locator_name) != 0) {
+	if (strcmp(bgp->srv6_locator_name, chunk->locator_name) != 0) {
 		zlog_err("%s: Locator name unmatch %s:%s", __func__,
-			 bgp->srv6_locator_name, s6c.locator_name);
+			 bgp->srv6_locator_name, chunk->locator_name);
 		return 0;
```

Comment:
You have to free the chunk here to avoid memory leak.

Reply:
Thanks. I fixed it.

Reply:
You have to use `srv6_locator_chunk_free`, not `free`. It is needed to correctly calculate memory statistics.

Reply:
I see. Thanks.